test/tests.spec.ts

```diff
@@ -1450,135 +1458,206 @@ describe("node-saml /", function () {
-        const base64xml = Buffer.from(xml).toString("base64");
-        const container = { SAMLResponse: base64xml };
-
-        const samlConfig = {
-          entryPoint: "https://app.onelogin.com/trust/saml2/http-post/sso/371755",
-          cert: "MIICrjCCAZYCCQDWybyUsLVkXzANBgkqhkiG9w0BAQsFADAZMRcwFQYDVQQDFA5hY21lX3Rvb2xzLmNvbTAeFw0xNTA4MTgwODQ3MzZaFw0yNTA4MTcwODQ3MzZaMBkxFzAVBgNVBAMUDmFjbWVfdG9vbHMuY29tMIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAlyT+OzEymhaZFNfx4+HFxZbBP3egvcUgPvGa7wWCV7vyuCauLBqwO1FQqzaRDxkEihkHqmUz63D25v2QixLxXyqaFQ8TxDFKwYATtSL7x5G2Gww56H0L1XGgYdNW1akPx90P+USmVn1Wb//7AwU+TV+u4jIgKZyTaIFWdFlwBhlp4OBEHCyYwngFgMyVoCBsSmwb4if7Mi5T746J9ZMQpC+ts+kfzley59Nz55pa5fRLwu4qxFUv2oRdXAf2ZLuxB7DPQbRH82/ewZZ8N4BUGiQyAwOsHgp0sb9JJ8uEM/qhyS1dXXxjo+kxsI5HXhxp4P5R9VADuOquaLIo8ptIrQIDAQABMA0GCSqGSIb3DQEBCwUAA4IBAQBW/Y7leJnV76+6bzeqqi+buTLyWc1mASi5LVH68mdailg2WmGfKlSMLGzFkNtg8fJnfaRZ/GtxmSxhpQRHn63ZlyzqVrFcJa0qzPG21PXPHG/ny8pN+BV8fk74CIb/+YN7NvDUrV7jlsPxNT2rQk8G2fM7jsTMYvtz0MBkrZZsUzTv4rZkF/v44J/ACDirKJiE+TYArm70yQPweX6RvYHNZLSzgg4o+hoyBXo5BGQetAjmcIhC6ZOwN3iVhGjp0YpWM0pkqStPy3sIR0//LZbskWWlSRb0fX1c4632Xb+zikfec4DniYV6CxkB2U+plHpOX1rt1R+UiTEIhTSXPNt/",
-          validateInResponseTo: false,
-        };
-        const samlObj = new SAML(samlConfig);
-
-        fakeClock = sinon.useFakeTimers(Date.parse("2014-06-05T12:07:07.662Z"));
-
-        // Mock the SAML request being passed through Passport-SAML
-        await samlObj.cacheProvider.saveAsync(requestId, new Date().toISOString());
+      );
 
-        const { profile } = await samlObj.validatePostResponseAsync(container);
-        expect(profile!.nameID!.startsWith("UIS/jochen-work")).to.be.true;
-        const value = await samlObj.cacheProvider.getAsync(requestId);
-        expect(value).to.exist;
-        expect(value!).to.equal("2014-06-05T12:07:07.662Z");
+      [false, ValidateInResponseTo.never].forEach((validateInResponseTo) => {
+        describe(`when validateInResponseTo is set to ${validateInResponseTo}`, () => {
+          it("onelogin xml document with InResponseTo not in the cache should validate", async () => {
```

Comment:
Correct me if I'm wrong, but this is the only new test; the rest are just wrapped for the values of `ValidateInResponseTo` that apply to them, correct?

Reply:
Ok, if this is the only new test, then I think your tests look good.

Reply:
Yes, you are right. Other modifications are just updates of existing tests to check that options are working with new values.